sql/src/test/java/io/crate/testing/UseSemiJoins.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.testing;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Execute "set semi_joins = true" before the actual statement to allow IN/ANY (Subquery)
+ * to be rewritten to a SEMI/ANTI join.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Inherited
+public @interface UseSemiJoins {
+
+    // 0 -> disabled
+    // 1 -> always enabled
+    // close to 1 -> mostly enabled
+    // close to 0 -> mostly disabled
+    double value() default 0.5;
```

Comment:
Do we ever use these float values? Why don't we have only `DISABLED`, `ENABLED`, and `RANDOM`?

Reply:
Hm actually never, just kept it consistent with @UseJdbc, should I change to the 3 values?

Reply:
Would be nice!

Reply:
we adapted the float value pattern from ES. No strong opinion but I'd stick with it for consistency and because I think it's easy to understand and the implementation is as simple as with an enum